src/resspect/time_domain_loop.py

```diff
@@ -739,64 +657,58 @@ def process_next_day_loop(
+    config
+        The `TimeDomainConfiguration` dataclass that
+        contains all the runtime configuration options.
     """
-    next_day_data = _load_next_day_data(
-        next_day_features_file_name, is_separate_files, is_queryable,
-        survey_name, ia_training_fraction, is_save_samples,
-        feature_extraction_method=feature_extraction_method)
+    next_day_data = _load_next_day_data(next_day_features_file_name, config)
     for metadata_value in light_curve_data.train_metadata[id_key_name].values:
         next_day_pool_metadata = next_day_data.pool_metadata[id_key_name].values
         if metadata_value in next_day_pool_metadata:
-            next_day_pool_metadata_indices = list(
-                next_day_pool_metadata).index(metadata_value)
+            next_day_pool_metadata_indices = list(next_day_pool_metadata).index(metadata_value)
             if metadata_value not in light_curve_train_ids:
-                light_curve_train_metadata = light_curve_data.train_metadata[
-                    id_key_name].values
+                light_curve_train_metadata = light_curve_data.train_metadata[id_key_name].values
                 light_curve_data = _remove_old_training_features(
-                    light_curve_data, light_curve_train_metadata,
-                    metadata_value)
+                    light_curve_data,
+                    light_curve_train_metadata,
+                    metadata_value
+                )
                 if light_curve_data.queryable_ids.shape[0] > 0:
                     light_curve_data = _update_queried_sample(
-                        light_curve_data, next_day_data, id_key_name,
-                        metadata_value)
+                        light_curve_data,
+                        next_day_data,
+                        id_key_name,
+                        metadata_value
+                    )
                 light_curve_data = _update_training_data_with_new_features(
-                    light_curve_data, next_day_data, metadata_value, id_key_name)
+                    light_curve_data,
+                    next_day_data,
+                    metadata_value,
+                    id_key_name
+                )
             next_day_data = _update_next_day_pool_data(
-                next_day_data, next_day_pool_metadata_indices)
+                next_day_data,
+                next_day_pool_metadata_indices
+            )
         next_day_data = _update_next_day_val_and_test_data(
-            next_day_data, metadata_value, id_key_name)
+            next_day_data,
+            metadata_value,
+            id_key_name
+        )
     light_curve_data = _update_light_curve_data_for_next_epoch(
-        light_curve_data, next_day_data, canonical_data, is_queryable, strategy,
-        is_separate_files)
+        light_curve_data,
+        next_day_data,
+        canonical_data,
+        config.queryable,
+        config.strategy,
+        config.sep_files
+    )
```

Comment:
I know that you didn't introduce this, but oh my, this looks like a place where lots of bugs can hide. 😬